### PR TITLE
feat: bump `openai` package to v1.1

### LINKED ISSUE #355: Bump `openai` package to v1.1.0

### PR DESCRIPTION
**Pull request by dandansamax**
## Description

Bump `openai` package to v1.1. Change APIs usage and type system to fit the new version.

close #355

## Motivation and Context

Why is this change required? What problem does it solve?
If it fixes an open issue, please link to the issue here.
You can use the syntax `close #15213` if this solves the issue #15213

- [ ] I have raised an issue to propose this change ([required](https://github.com/camel-ai/camel/issues) for new features and bug fixes)

## Types of changes

What types of changes does your code introduce? Put an `x` in all the boxes that apply:

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds core functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Documentation (update in the documentation)
- [ ] Example (update in the folder of example)

## Implemented Tasks

- [ ] Subtask 1
- [ ] Subtask 2
- [ ] Subtask 3

## Checklist

Go over all the following points, and put an `x` in all the boxes that apply.
If you are unsure about any of these, don't hesitate to ask. We are here to help!

- [x] I have read the [CONTRIBUTION](https://github.com/camel-ai/camel/blob/master/CONTRIBUTING.md) guide. (**required**)
- [ ] My change requires a change to the documentation.
- [ ] I have updated the tests accordingly. (*required for a bug fix or a new feature*)
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: camel/agents/chat_agent.py [hunk 1]
```diff
@@ -368,22 +357,23 @@ def handle_batch_response(
             tuple: A tuple of list of output `ChatMessage`, list of
                 finish reasons, usage dictionary, and response id.
         """
+        response_dict = response.model_dump()
```
Comment (lightaime) [inline camel/agents/chat_agent.py:360]: Why `model_dump()` is necessary? Can we just loop `for choice in respoense.choices`?

**Review by lightaime — COMMENTED** [review]
Thanks @dandansamax!! Left some initial comments.

**Review by lightaime — APPROVED** [review]
LGTM!

File: examples/embodiment/code_execution.py
[the remark this answers is not in the document]
Comment (dandansamax) [reply examples/embodiment/code_execution.py:1]: These were commited by accident. Reverted.

**Pull request merged**